kernel/src/engine/sync/mod.rs

```diff
@@ -19,16 +19,15 @@ mod storage;
 
 /// This is a simple implementation of [`Engine`]. It only supports reading data from the local
```

Comment:
Let's add a note that this is a "test only" engine.

Also, my bad, but this whole comment should be a `//!` mod comment.